### PR TITLE
WRP PSP e_particles

### PR DESCRIPTION
**Pull request by Xeeynamo**
Quite a big one. I merged `EntitySoulStealOrb` across all overlays by normalising the symbol names. The symbol types was all over the place, so I had to fix that as well.

I solved the s16/u16 ifdef between PSX and PSP. The only part I couldn't match without a ifdef was `angle = (angle - 1) * 8;` and `angle = (angle * 8) - 8;`.

I also decompiled the exclusive PSP functions `func_psp_0923AD68` and `func_psp_0923B2F0` and merged the PSX and PSP `e_particles.c` code. The exclusive functions required a file split.

### CODE REVIEW

File: config/symbols.pspeu.stwrp.txt [hunk 1]
```diff
@@ -71,9 +71,9 @@ EntityWarpRoom = 0x09244d28;
 EntityWarpSmallRocks = 0x09245c50;
 g_eRedDoorUV = 0x09246080;
 UNK_Invincibility0 = 0x09246098;
-D_801810A0 = 0x092460C8;
-D_801810B0 = 0x092460D8;
-D_80181110 = 0x09246138;
+g_ESoulStealOrbAngles = 0x092460C8;
+g_ESouldStelOrbSprt = 0x092460D8;
```
Comment (sozud) [inline config/symbols.pspeu.stwrp.txt:75]: Is this spelling intentional? Sould / Stel

File: src/st/wrp_psp/e_particles_maria1.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
+#include "../wrp/wrp.h"
+
+// INCLUDE_ASM("st/wrp_psp/psp/wrp_psp/e_particles_maria1", func_psp_0923AD68);
```
Comment (sozud) [inline src/st/wrp_psp/e_particles_maria1.c:3]: Remove comment?

File: src/st/wrp_psp/e_particles_maria1.c [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            prim->drawMode = DRAW_HIDE;
+            self->flags |= FLAG_HAS_PRIMS;
+            self->primIndex = primIndex;
+            self->animSet = 0;
+            angle = GetAngleBetweenEntities(self, pl);
+            direction = 0;
+            if (self->posY.i.hi > 0x70) {
+                direction = 1;
+            }
+            if (self->posX.i.hi < *(s16*)0x091E1682) {
```
Comment (sozud) [inline src/st/wrp_psp/e_particles_maria1.c:32]: Do we not want a symbol for this?

**Review by sozud — APPROVED** [review]

File: config/symbols.pspeu.stwrp.txt (hunk 1 above)
Comment (Xeeynamo) [reply config/symbols.pspeu.stwrp.txt]: Done this and and the others, nice catch!

**Pull request merged**